packages/graphql/src/mutations/_txHelper.js

```diff
@@ -140,6 +142,437 @@ async function fetchGasPrice() {
+  }
+  // But use it when we're using our provider
+  return await w3.eth.getGasPrice()
+}
+
+/**
+ * ABI encode a transaction if necessary
+ *
+ * @param {object} a web3.js transaction object
+ * @returns {string} hex encoded transaction
+ */
+async function encodeTransaction(tx) {
+  if (
+    typeof tx !== 'object' ||
+    !Object.prototype.hasOwnProperty.call(tx, 'encodeABI')
+  ) {
+    return tx
+  }
+  return await tx.encodeABI()
+}
+
+/**
+ * Wrap a transaction for sending through a proxy
+ *
+ * @param {object} args
+ * @param {object} args.ProxyContract - An instance of IdentityProxy
+ * @param {object} args.tx - A web3.js transaction object
+ * @param {string} args.destinationContract - Address of the target contract
+ * @param {string} args.value - the value being sent with the tx
+ * @returns {object} a web3.js transaction object
+ */
+async function txWrapExecute({
+  ProxyContract,
+  tx,
+  destinationContract,
+  value
+}) {
+  const txData = await encodeTransaction(tx)
+
+  return ProxyContract.methods.execute(
+    0,
+    destinationContract,
+    value || '0',
+    txData
+  )
+}
+
+/**
+ * Wrap a transaction for sending through a proxy, and wrap that in a create
+ * proxy tx.
+ *
+ * @param {object} args
+ * @param {object} args.ProxyFactory - An instance of ProxyFactory
+ * @param {object} args.tx - A web3.js transaction object
+ * @param {string} args.from - The originating Ethereum account address
+ * @returns {object} a web3.js transaction object
+ */
+async function txWrapCreateProxy({ ProxyFactory, tx, from }) {
+  const txData = await encodeTransaction(tx)
+
+  return ProxyFactory.methods.createProxyWithSenderNonce(
+    contracts.config.IdentityProxyImplementation,
+    txData,
+    from,
+    '0'
+  )
+}
+
+/**
+ * Wrap a transaction for sending through a proxy, and wrap that in a create
+ * proxy tx that also sets owner.
+ *
+ * @param {object} args
+ * @param {object} args.ProxyContract - An instance of IdentityProxy
+ * @param {object} args.tx - A web3.js transaction object
+ * @param {string} args.from - The originating Ethereum account address
+ * @param {string} args.destinationContract - Address of the target contract
+ * @param {string} args.value - the value being sent with the tx
+ * @returns {object} a web3.js transaction object
+ */
+async function txWrapChangeOwner({
+  ProxyContract,
+  tx,
+  from,
+  destinationContract,
+  value
+}) {
+  const txData = await encodeTransaction(tx)
+
+  return ProxyContract.methods.changeOwnerAndExecute(
+    from,
+    destinationContract,
+    value || '0',
+    txData
+  )
+}
+
+/**
+ * Run callback functions giving them val as an argument
+ *
+ * @param {object} args
+ * @param {Array} args.callbacks - Array of functions to call with val
+ * @param {T} args.val - Value to provide callbacks as an argument
+ * @returns {undefined}
+ */
+async function handleCallbacks({ callbacks, val }) {
+  if (!callbacks) return
+  for (let i = 0; i < callbacks.length; i++) {
```

Comment:
The behavior of things like `forEach` and `map` get really weird with async callbacks.  They don't await, when I explicitly wanted blocking here.

Reply:
We could also use the "for of" syntax:

     for (const callback of callbacks) {
         const ret = await callback()
     }

Reply:
Always mix up `for ... of` and `for .. in` when using JS so I tend to lean on the old school method instead of looking it up every time.  Will include this in my cleanup PR though, thanks.